plugin/winresizer.vim

```diff
@@ -74,71 +103,137 @@ let s:codeList = {
+        \  'mode' : g:winresizer_keycode_mode,
         \}
 
 exe 'nnoremap ' . g:winresizer_start_key .' :WinResizerStartResize<CR>'
 
-com! WinResizerStartResize call WinResizerStartResize()
+com! WinResizerStartResize call s:startReize(s:resizeCommands())
+com! WinResizerStartMove call s:startReize(s:moveCommands())
+com! WinResizerStartFocus call s:startReize(s:focusCommands())
 
 if has("gui_running") && g:winresizer_gui_enable != 0
   exe 'nnoremap ' . g:winresizer_gui_start_key .' :WinResizerStartResizeGUI<CR>'
-  com! WinResizerStartResizeGUI call WinResizerStartResizeGUI()
+  com! WinResizerStartResizeGUI call s:startReize(s:guiResizeCommands())
 endif
 
-fun! WinResizerStartResizeGUI()
-  if g:winresizer_enable == 0
-    return
-  endif
+fun! s:guiResizeCommands()
+
   let commands = {
+      \  'mode'   : "resize",
       \  'left'   : 'let &columns = &columns - ' . g:winresizer_vert_resize,
       \  'right'  : 'let &columns = &columns + ' . g:winresizer_vert_resize,
       \  'up'     : 'let &lines = &lines - ' . g:winresizer_horiz_resize,
       \  'down'   : 'let &lines = &lines + ' . g:winresizer_horiz_resize,
       \  'cancel' : 'let &columns = ' . &columns . '|let &lines = ' . &lines . '|',
       \}
-  call s:startReize(commands)
+
+  return commands
 
 endfun
 
-fun! WinResizerStartResize()
-  if g:winresizer_enable == 0
-    return
-  endif
+fun! s:tuiResizeCommands()
 
   let behavior = s:getResizeBehavior()
 
   let commands = {
+        \  'mode'   : "resize",
         \  'left'   : ':vertical resize ' . behavior['left']  . g:winresizer_vert_resize,
         \  'right'  : ':vertical resize ' . behavior['right'] . g:winresizer_vert_resize,
         \  'up'     : ':resize ' . behavior['up']   . g:winresizer_horiz_resize,
         \  'down'   : ':resize ' . behavior['down'] . g:winresizer_horiz_resize,
         \  'cancel' : winrestcmd(),
         \}
 
-  call s:startReize(commands)
+  return commands
+
+endfun
+
+fun! s:resizeCommands()
+  if has("gui_running") && g:winresizer_gui_enable != 0
+    return s:guiResizeCommands()
+  else
+    return s:tuiResizeCommands()
+  endif
+endfun
+
+fun! s:moveCommands()
+
+  let commands = {
+        \  'mode'   : "move",
+        \  'left'   : ":call winresizer#swapTo('left')",
+        \  'right'  : ":call winresizer#swapTo('right')",
+        \  'up'     : ":call winresizer#swapTo('up')",
+        \  'down'   : ":call winresizer#swapTo('down')",
+        \  'cancel' : winrestcmd(),
+        \}
+
+  return commands
+
+endfun
+
+fun! s:focusCommands()
+
+  let commands = {
+        \  'mode'   : "focus",
+        \  'left'   : "normal! \<c-w>h",
+        \  'right'  : "normal! \<c-w>l",
+        \  'up'     : "normal! \<c-w>k",
+        \  'down'   : "normal! \<c-w>j",
+        \  'cancel' : winrestcmd(),
+        \}
+
+  return commands
 
 endfun
 
 fun! s:startReize(commands)
 
+  if g:winresizer_enable == 0
+    return
+  endif
+
+  let s:commands = a:commands
```

Comment:
Why did you choice `s:commands` ?
I think `let commands` (It's equal `let l:commands`) is better.